stacks-signer/src/v0/signer.rs

```diff
@@ -617,6 +667,85 @@ impl Signer {
     }
 
+    /// Check the current tracked submitted block proposal to see if it has timed out.
+    /// Broadcasts a rejection and marks the block locally rejected if it has.
+    fn check_submitted_block_proposal(&mut self) {
+        let Some((block_proposal, block_submission)) = self.submitted_block_proposal.clone() else {
+            // Nothing to check.
+            return;
+        };
+        if block_submission.elapsed() < self.block_proposal_validation_timeout {
+            // Not expired yet.
+            return;
+        }
+        // Let us immediately flush, even if we later encounter an error broadcasting our responses.
+        // We should still attempt to handle a new proposal at this point.
+        self.submitted_block_proposal = None;
+        let signature_sighash = block_proposal.block.header.signer_signature_hash();
+        // For mutability reasons, we need to take the block_info out of the map and add it back after processing
+        let mut block_info = match self
+            .signer_db
+            .block_lookup(self.reward_cycle, &signature_sighash)
+        {
+            Ok(Some(block_info)) => {
+                if block_info.state == BlockState::GloballyRejected
+                    || block_info.state == BlockState::GloballyAccepted
+                {
+                    // The block has already reached consensus. This should never really hit, but check just to be safe.
+                    self.submitted_block_proposal = None;
+                    return;
+                }
+                block_info
+            }
+            Ok(None) => {
+                // This is weird. If this is reached, its probably an error in code logic or the db was flushed.
+                // Why are we tracking a block submission for a block we have never seen / stored before.
+                error!("{self}: tracking an unknown block validation submission.";
+                    "signer_sighash" => %signature_sighash,
+                    "block_id" => %block_proposal.block.block_id(),
+                );
+                self.submitted_block_proposal = None;
+                return;
+            }
+            Err(e) => {
+                error!("{self}: Failed to lookup block in signer db: {e:?}",);
+                self.submitted_block_proposal = None;
+                return;
+            }
+        };
+        warn!(
+            "{self}: Failed to receive block validation response within {} ms. Rejecting block.", self.block_proposal_validation_timeout.as_millis();
+            "signer_sighash" => %signature_sighash,
+            "block_id" => %block_proposal.block.block_id(),
+        );
+        let rejection = BlockResponse::rejected(
+            block_proposal.block.header.signer_signature_hash(),
+            RejectCode::ConnectivityIssues,
+            &self.private_key,
+            self.mainnet,
+        );
+        // We know proposal is invalid. Send rejection message, do not do further validation
```

Comment:
Was this comment copied from somewhere? It doesn't seem right here.

Reply:
Yep. Copy paste leftover. Fixed.